packages/frontend/src/ee/features/aiCopilot/components/ChatElements/AiChartVisualization.tsx

```diff
@@ -1,24 +1,122 @@
 
-type Props = {
-    vizData: ReturnType<typeof getChartVisualizationFromAiQuery>;
+import { type EchartSeriesClickEvent } from '../../../../../components/SimpleChart';
+import { type EChartSeries } from '../../../../../hooks/echarts/useEchartsCartesianConfig';
+import useHealth from '../../../../../hooks/health/useHealth';
+import { useOrganization } from '../../../../../hooks/organization/useOrganization';
+import { useExplore } from '../../../../../hooks/useExplore';
+import { type InfiniteQueryResults } from '../../../../../hooks/useQueryResults';
+import { getChartOptionsFromAiAgentThreadMessageVizQuery } from '../../utils/echarts';
+import { getAiAgentChartConfig } from '../../utils/getAiAgentChartConfig';
+
+type Props = ApiAiAgentThreadMessageVizQuery & {
+    vizConfig: AiAgentMessageAssistant['vizConfigOutput'];
+    results: InfiniteQueryResults;
 };
 
-export const AiChartVisualization: FC<Props> = ({ vizData }) => {
-    if (!vizData) return null;
+export const AiChartVisualization: FC<Props> = ({
+    query,
+    results,
+    type,
+    vizConfig,
+}) => {
+    const { data: health } = useHealth();
+    const { data: organization } = useOrganization();
+    const { metricQuery, fields } = query;
+    const tableName = metricQuery?.exploreName;
+    const { data: explore } = useExplore(tableName);
+    const [echartsClickEvent, setEchartsClickEvent] =
+        useState<EchartSeriesClickEvent | null>(null);
+    const [echartSeries, setEchartSeries] = useState<EChartSeries[]>([]);
+
+    const resultsData = useMemo(
+        () => ({
+            ...results,
+            metricQuery,
+            fields,
+        }),
+        [results, metricQuery, fields],
+    );
+
+    const chartOptions = getChartOptionsFromAiAgentThreadMessageVizQuery({
+        config: vizConfig,
+        rows: results.rows,
+        type,
+    });
+    const chartConfig = getAiAgentChartConfig({
+        type: type,
+        chartOptions,
+        metricQuery,
+    });
 
     return (
         <Box h="100%" mih={400}>
-            <VisualizationProvider minimal {...vizData}>
-                <LightdashVisualization
-                    className="sentry-block ph-no-capture"
-                    data-testid="ai-visualization"
-                />
-            </VisualizationProvider>
+            <MetricQueryDataProvider
+                metricQuery={metricQuery}
+                tableName={tableName}
+                explore={explore}
+                queryUuid={query.queryUuid}
+            >
+                <VisualizationProvider
+                    resultsData={resultsData}
+                    chartConfig={chartConfig}
+                    columnOrder={[
+                        ...metricQuery.dimensions,
+                        ...metricQuery.metrics,
+                    ]}
+                    pivotTableMaxColumnLimit={
+                        health?.pivotTable.maxColumnLimit ?? 60 // TODO :: do we have a default?
```

Comment:
You can move up to 3 dimensions as columns and there is a limit of 60 column values in your table (e.g. pivoting by a dimension with 30 months, and another with 4 statuses won’t work because that will generate 4 x 30 = 120 column values).

https://docs.lightdash.com/references/chart-types#pivot-tables-dimensions-as-columns

Reply:
https://github.com/lightdash/lightdash/blob/e1c45cfb6a139b7db2390bbd1a6a92a2d30e1049/packages/backend/src/config/parseConfig.ts#L1066-L1071